charts/linkerd-control-plane/templates/destination.yaml

```diff
@@ -208,10 +208,15 @@ spec:
         {{- range (.Values.destinationController).experimentalArgs }}
         - {{ . }}
         {{- end }}
-        {{- with (.Values.destinationController).experimentalEnv }}
+        {{- if or (.Values.destinationController).additionalEnv (.Values.destinationController).experimentalEnv }}
         env:
+        {{- with (.Values.destinationController).additionalEnv }}
         {{- toYaml . | nindent 8 -}}
         {{- end }}
+        {{- with (.Values.destinationController).experimentalEnv }}
+        {{- toYaml . | nindent 8 -}}
+        {{- end }}
+        {{- end }}
```

Comment:
tioli, I think we could minimize this with something like
```
        {{- with .Values.destinationController | concat .experimentalEnv .additionalEnv }}
```

I haven't tested this specific syntax but the general idea should avoid duplication.  Here and throughout.  Furthermore, we could consider using a partial here to reduce boilerplate

Reply:
It appears `with` cannot be used in a pipe. So we have to replicate `.Values.destinationController` on the right. And concat will error if one of its args is nil, so we have to pipe to `default list ""` both vars. So the one liner ends up being not pretty.